drivers/shared/executor/executor_linux.go

```diff
@@ -682,9 +682,16 @@ func configureCgroups(cfg *lconfigs.Config, command *ExecCommand) error {
 		return nil
 	}
 
-	if mb := command.Resources.NomadResources.Memory.MemoryMB; mb > 0 {
-		// Total amount of memory allowed to consume
-		cfg.Cgroups.Resources.Memory = mb * 1024 * 1024
+	// Total amount of memory allowed to consume
+	var memoryLimit int64
+	if res := command.Resources.NomadResources.Memory; res.MemoryMaxMB > 0 {
+		memoryLimit = res.MemoryMaxMB * 1024 * 1024
+	} else if res.MemoryMB > 0 {
+		memoryLimit = res.MemoryMB * 1024 * 1024
+	}
```

Comment:
Initially I intended to have executor use the linux resources value - but opted to "reimplement" the conditional here to avoid updating many tests - current exec tests don't seem to populate linux resources. Just laziness, and can update them with a bit of a nudge.

Reply:
This executor gets used (correctly or not) as a model for other third-party executors. IIRC `containerd` was based heavily on this. We probably should do whatever we expect other folks to do here; if we don't use the Linux resources value, it's less clear what our intent for that field is.

Reply:
Excellent point!  Laziness is totally the wrong thing here.  Will update! thanks.